exchanges/poloniex/poloniex.go

```diff
@@ -3,933 +3,1113 @@ package poloniex
+	if err := e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, sCancelAllSmartOrdersEPL, http.MethodDelete, "/smartorders", nil, args, &resp); err != nil {
+		return nil, fmt.Errorf("%w: %w", order.ErrCancelFailed, err)
+	}
+	return resp, nil
 }
 
-// GetLendingHistory returns lending history for the account
-func (e *Exchange) GetLendingHistory(ctx context.Context, start, end string) ([]LendingHistory, error) {
-	vals := url.Values{}
-
-	if start != "" {
-		vals.Set("start", start)
+func orderFillParams(arg *OrdersHistoryRequest) (url.Values, error) {
+	params := url.Values{}
+	if arg.AccountType != "" {
+		params.Set("accountType", arg.AccountType)
 	}
-
-	if end != "" {
-		vals.Set("end", end)
+	if arg.OrderType != "" {
+		params.Set("type", arg.OrderType)
 	}
-
-	var resp []LendingHistory
-	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodPost,
-		poloniexLendingHistory,
-		vals,
-		&resp)
+	if len(arg.OrderTypes) > 0 {
+		params.Set("types", strings.Join(arg.OrderTypes, ","))
+	}
+	if arg.Side != order.UnknownSide && arg.Side != order.AnySide {
+		params.Set("side", arg.Side.String())
+	}
+	if !arg.Symbol.IsEmpty() {
+		params.Set("symbol", arg.Symbol.String())
+	}
+	if arg.From > 0 {
+		params.Set("from", strconv.FormatInt(arg.From, 10))
+	}
+	if arg.Direction != "" {
+		params.Set("direction", arg.Direction)
+	}
+	if arg.States != "" {
+		params.Set("states", arg.States)
+	}
+	if arg.Limit > 0 {
+		params.Set("limit", strconv.FormatInt(arg.Limit, 10))
+	}
+	if arg.HideCancel {
+		params.Set("hideCancel", "true")
+	}
+	if !arg.StartTime.IsZero() && !arg.EndTime.IsZero() {
+		if err := common.StartEndTimeCheck(arg.StartTime, arg.EndTime); err != nil {
+			return nil, err
+		}
+	}
+	if !arg.StartTime.IsZero() {
+		params.Set("startTime", strconv.FormatInt(arg.StartTime.UnixMilli(), 10))
+	}
+	if !arg.EndTime.IsZero() {
+		params.Set("endTime", strconv.FormatInt(arg.EndTime.UnixMilli(), 10))
+	}
+	return params, nil
 }
 
-// ToggleAutoRenew allows for the autorenew of a contract
-func (e *Exchange) ToggleAutoRenew(ctx context.Context, orderNumber int64) (bool, error) {
-	values := url.Values{}
-	values.Set("orderNumber", strconv.FormatInt(orderNumber, 10))
-	result := GenericResponse{}
-
-	err := e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodPost,
-		poloniexAutoRenew,
-		values,
-		&result)
+// GetOrdersHistory gets a list of historical orders in an account
+func (e *Exchange) GetOrdersHistory(ctx context.Context, arg *OrdersHistoryRequest) ([]*TradeOrder, error) {
+	params, err := orderFillParams(arg)
 	if err != nil {
-		return false, err
+		return nil, err
 	}
-
-	if result.Success == 0 {
-		return false, errors.New(result.Error)
+	var resp []*TradeOrder
+	if err := e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, sGetOrderHistoryEPL, http.MethodGet, "/orders/history", params, nil, &resp); err != nil {
+		return nil, fmt.Errorf("%w %v", order.ErrGetFailed, err)
```

Comment:
There is an issue with this function for  `exchange_wrapper_standards_test.go` which needs fixing.
```
    exchange_wrapper_standards_test.go:229: Poloniex Func 'GetOrderHistory' Error: 'get order failed authenticated request failed Poloniex unsuccessful HTTP status code: 400 raw response: {"message":"Invalid Apikey or Signature","code":400}, authenticated request failed'. Inputs: [<context> {[ETH-TEST-DELIM-usdt] spot ANY ANY  0001-01-01 00:00:00 +0000 UTC 0001-01-01 00:00:00 +0000 UTC 1337 }].
--- FAIL: TestAllExchangeWrappers/poloniex_wrapper_tests/GetOrderHistory-spot-ETH-TEST-DELIM-usdt (0.16s)
```

This kind of issue should be caught by returning a wrapped error `request.ErrAuthRequestFailed` or being sent as ` request.AuthenticatedRequest`

Please double check. This is the only function which is having the issue. Just to repeat, it is okay that it returns the error "invalid api key", but it should still pass in exchange_wrapper_standards.go

I believe it is likely due to the following code:
```suggestion
		return nil, fmt.Errorf("%w %w", order.ErrGetFailed, err)
```